sui-execution/latest/sui-move-natives/src/crypto/sui_state_proof.rs

```diff
@@ -0,0 +1,287 @@
+        Err(e) => return Ok(NativeResult::err(cost, INVALID_TX)),
+    }
+
+    // println!("giga debug 4");
+
+    let next_committee_epoch;
+    // Extract the new committee information
+    if let Some(EndOfEpochData {
+        next_epoch_committee,
+        ..
+    }) = &checkpoint_summary.end_of_epoch_data
+    {
+        let next_committee = next_epoch_committee.iter().cloned().collect();
+        next_committee_epoch =
+            Committee::new(checkpoint_summary.epoch().checked_add(1).unwrap(), next_committee);
+    } else {
+        return Ok(NativeResult::err(cost, INVALID_TX))
+    }
+
+    // println!("giga debug 5");
+
+    Ok(NativeResult::ok(cost, smallvec![Value::vector_u8(bcs::to_bytes(&next_committee_epoch).unwrap())]))
+}
+
+
+
+
+
+/***************************************************************************************************
+ * native fun dkg_verify_decommitment_and_proof_of_centralized_party_public_key_share
+ * Implementation of the Move native function `dwallet_2pc_mpc_ecdsa_k1::dkg_verify_decommitment_and_proof_of_centralized_party_public_key_share(commitment_to_centralized_party_secret_key_share: vector<u8>, secret_key_share_encryption_and_proof: vector<u8>, centralized_party_public_key_share_decommitment_and_proofs: vector<u8>): (vector<u8>, vector<u8>, vector<u8>);`
+ *   gas cost: dkg_verify_decommitment_and_proof_of_centralized_party_public_key_share_cost_base   | base cost for function call and fixed opers
+ **************************************************************************************************/
+ pub fn sui_state_proof_verify_transaction(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 6);
+
+    // Load the cost parameters from the protocol config
+    let sui_state_proof_cost_params = &context
+        .extensions()
+        .get::<NativesCostTable>()
+        .twopc_mpc_dkg_cost_params // TODO
+        .clone();
+
+
+    // Load the cost parameters from the protocol config
+    // let object_runtime = context
+    //     .extensions()
+    //     .get::<ObjectRuntime>();
+
+    // Charge the base cost for this oper
+    native_charge_gas_early_exit!(
+        context,
+        sui_state_proof_cost_params.dkg_verify_decommitment_and_proof_of_centralized_party_public_key_share_cost_base
+    );
+
+    let cost = context.gas_used();
+
+
+    let package_id_bytes = pop_arg!(args, Vec<u8>);
+    let type_layout_bytes = pop_arg!(args, Vec<u8>);
+    let transaction_bytes = pop_arg!(args, Vec<u8>);
+    let checkpoint_contents_bytes = pop_arg!(args, Vec<u8>);
+    let summary_bytes = pop_arg!(args, Vec<u8>);
+    let committee_bytes = pop_arg!(args, Vec<u8>);
+
+    let Ok(committee) = bcs::from_bytes::<Committee>(&committee_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_COMMITTEE
+        ));
+    };
```

Comment:
I'm a little confused. The `Committee` struct, where is that defined, who is creating that?
Is it a Rust or Move struct?